### PR TITLE
Second attempt to add Microsoft.NETCore.Native package

### PR DESCRIPTION
**Pull request by SedarG**
This time I'm including a fix for the osx release build.
@jkotas , could you please take a look
/cc @dotnet/corert-contrib

### CODE REVIEW

File: src/.nuget/Microsoft.NETCore.Native/Microsoft.NETCore.Native.builds [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+  <Import Project="$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildThisFileDirectory), dir.props))\dir.props" />
+
+  <PropertyGroup>
+    <!-- This property must be set to the same value as $(PackageOutputPath) for the nuspecs and nupkgs to be binplaced to the intended location. -->
+    <OutputPath>$(PackageOutputPath)</OutputPath>
+  </PropertyGroup>
+
+  <!-- We always build the identity/redirection package. However, the platform specific runtime-*.nupkg is built based upon the target OS we are building the product for. -->
+  <ItemGroup>
+    <Project Include="Microsoft.NETCore.Native.pkgproj">
```
Comment (jkotas) [inline src/.nuget/Microsoft.NETCore.Native/Microsoft.NETCore.Native.builds]: Why do we call this Microsoft.NETCore.Native? The similar packages in corefx are called runtime.native.System.IO.Compression, native.System.Security.Cryptography. Shouldn't this one be called runtime.native.System.Globalization to follow the same naming scheme?

Comment (jkotas) [inline src/.nuget/Microsoft.NETCore.Native/Microsoft.NETCore.Native.builds]: (I should have thought about this in the original review...)

**Comment by jkotas** [comment]
LGTM otherwise.

File: src/.nuget/Microsoft.NETCore.Native/Microsoft.NETCore.Native.builds (hunk 1 above)
Comment (SedarG) [inline src/.nuget/Microsoft.NETCore.Native/Microsoft.NETCore.Native.builds]: runtime.xxx.PackageIdentity is the scheme that we generally use in CoreFX for packages that contain the runtime-specific-implementations of the library packages. I'd guess runtime.native.XXX tries to follow that approach, unnecessarily. IMHO, adding runtime.native is prefix just makes the name longer without much benefit. Thinking about it again, Microsoft.NETCore.Native is not the best name either, but at least it's friendlier. 
And it's similar to the other packages in CoreCLR repo, which are structurally similar to this one, such as Microsoft.NETCore.JIT.

Comment (jkotas) [inline src/.nuget/Microsoft.NETCore.Native/Microsoft.NETCore.Native.builds]: > runtime.native.XXX tries to follow that approach, unnecessarily

Naming tends to be imperfect, but it is very valuable to be consistently imperfect. 

> And it's similar to the other packages in CoreCLR repo, which are structurally similar to this one, such as Microsoft.NETCore.JIT.

I do not think that this package is structurally similar to JIT.

Comment (SedarG) [inline src/.nuget/Microsoft.NETCore.Native/Microsoft.NETCore.Native.builds]: For now I'll keep the name as is and we'll come back when the CoreFX changes on the packaging gets settled.

**Pull request merged**